ci/gpu/build.sh

```diff
@@ -81,6 +86,8 @@ else
     logger "Check GPU usage..."
     nvidia-smi
 
+    mkdir -p "$LIBCUDF_KERNEL_CACHE_PATH" || echo "could not mkdir -p $LIBCUDF_KERNEL_CACHE_PATH"
```

Comment:
I wouldn't expect I'd need to create this path first, but seems like it helps intermittent `permission denied` errors running the tests?

Reply:
This is expected. If you don't create the path first and you try to add files of any sort to the path you will get permission denied as the folder to write to doesn't exist.

Reply:
Yeah that would make sense, except the C++ function calls `boost::filesystem::create_directories()` every time the `getCacheDir()` function is called:
https://github.com/rapidsai/cudf/blob/14f8b785d1ddc1d9593ace0d968f3f8c2bd39874/cpp/src/jit/cache.cpp#L51-L61

But I just noticed one of the legacy JIT-cache tests removes the cache dir on setup/teardown:
https://github.com/rapidsai/cudf/blob/11fca091927777d48c6ee74ab8552279dc11313e/cpp/tests/binaryop/legacy/unit/jit-cache-test.hpp#L36-L50

I'm not sure if this is still sound with nested subdirs under `/tmp`. Prior to this PR the JIT-cache path was always something like `/tmp/cudf_0.12.0`. Now there's a unique subdir per CI job under `/tmp/cudf_0.12.0/<unique-ci-job-name>`.

Here's the scenario I'm worried about:
1. `CI Job 1` does `mkdir -p /tmp/cudf_0.12.0/ci-job-1`
  a. `/tmp/cudf_0.12.0` is created with `CI Job 1`'s user/group roles (or the NFS equivalent?)
2. `CI Job 2` does `mkdir -p /tmp/cudf_0.12.0/ci-job-2`
  a. permission denied because `/tmp/cudf_0.12.0` is owned/grouped by `CI Job 1`

Reply:
I just pushed [a commit](https://github.com/rapidsai/cudf/pull/3507/commits/4890a459aa61c60cae8bc2e422325a42724a8243) in #3507 . that tweaks the `build.sh` script a bit to remove the intermediate `cudf_0.12.0` directory in order to test this hypothesis. Will check the results in the AM.

Reply:
All of the CI jobs are run by user `jenkins` with the same UID and GID, which is why I'm initially confused as to why you're getting permissions error in the first place. I understand what is ultimately happening but I'm still trying to root cause it.